### ISSUE TITLE
Memory leak regression after upgrading to `>= 6.3.0`

### ISSUE THREAD

**Issue by johanrd** [ember-intl/ember-intl#1848]
Hi. I have experienced memory leaks in my app after upgrading ember-intl from `v6.2.2` to `>v6.3.0`.

**Steps to reproduce:** 

1) Clone https://github.com/johanrd/ember-memory-leak
2) Check out the `ember-intl` branch
3) Run `pnpm start`
4) Transition between `about` and `dashboard` a few times, and see that e.g. the ChartJs Component piles up in memory each time you enter the `dashboard` route:

https://github.com/ember-intl/ember-intl/assets/4601554/21c7aaa6-2d5c-4c37-9dca-44763a5ec821

5) Downgrade to ember-intl `v6.2.2` and see that the memory leaks with the ChartJs Component are gone:

https://github.com/ember-intl/ember-intl/assets/4601554/7fa4c89f-2d31-47d8-89aa-0da3b10ade08

**Comment by ijlee2** [comment]
Hi, @johanrd. Thanks for posting recordings to show some of the steps that you had taken. From the videos, though, I didn't understand what I'm supposed to look at, which suggests that there is a memory leak. Can you describe this to me?

Secondly, I found the [commit history](https://github.com/johanrd/ember-memory-leak/commits) hard to follow. Can you recreate the history so that each commit (each step) clearly indicates what's particular about that step?

At the moment, I have to doubt that `ember-intl` is the issue, because your reproduction repo includes a few other variables, such as `ember-truth-helpers` (in whose repo [you had raised a similar issue](https://github.com/jmurphyau/ember-truth-helpers/issues/205)), `ember-resources`, `ember-template-imports` (for `*.gts`), `ember-route-template`, and `chart.js`.

```json5
/* package.json */
{
  "dependencies": {
    "@ember/render-modifiers": "^2.1.0",
    "chart.js": "^4.4.2",
    "ember-intl": "^6.5.1",
    "ember-resources": "^7.0.0",
    "ember-route-template": "^1.0.3",
    "ember-truth-helpers": "^4.0.3",
    "reactiveweb": "^1.2.2"
  }
}
```

If you can write a reproduction app with the minimum necessary dependencies, I think it will help with your investigation more.

**Comment by johanrd** [comment]
@ijlee2 Hi. Thanks for the quick reply.  Yes, the commit history is a mess – _memory leaks are a mess._

Don't follow the commit history. **Please follow my steps to reproduce instead**

1) Clone https://github.com/johanrd/ember-memory-leak
2) Check out the `ember-intl` branch
3) Run `pnpm start`
4) Click on the "about' link.
5) Take a heap snapshot.
6) Search for "ChartJs". See that there are no such component in memory.
7) Visit the "dashboard" page and go back to "about" N times.
8) Take a new snapshot. See that the ChartJS component is retained in memory for each visit to `dashboard`, i.e. N times
<img width="732" alt="Screenshot_2024-03-08_at_13_31_47" src="https://github.com/ember-intl/ember-intl/assets/4601554/6379507b-33ea-4747-bde2-4c3e0760e098">

9) **Downgrade to `ember-intl v6.2.2.`**
10) Repeat steps 1-7 and see that ChartJS component is successfully released from memory every time you exit the dashboard route.

**Comment by johanrd** [comment]
Also if you remove this line  https://github.com/johanrd/ember-memory-leak/blob/ember-intl/app/templates/dashboard.gjs#L56 (an arg solely dependent on ember-intl), the memory leak is gone. the `@title={{t 'Character heights'}} isn't even _consumed_ by the ChartJS component, yet it makes the comonent stay in memory after teardown.

**Comment by johanrd** [comment]
Try again now.

I stripped it down to a very minimally reproducible example here: https://github.com/johanrd/ember-memory-leak
<img width="522" alt="Screenshot 2024-03-08 at 14 02 03" src="https://github.com/ember-intl/ember-intl/assets/4601554/c11fabdc-1b3f-4ec9-aacb-de36bed52771">

Just search for `LeakyComponent` instead of chart js, and it is evident that the comonent is not released from memory when it has any binding to a value from the t helper, even if it does not consume it.

Downgrade to ember-intl v6.2.2 and the memory leak is gone
